dictionary.go

```diff
@@ -296,3 +314,118 @@ func (a *adder) addWordUnknownWord(w string, countable bool) {
 		a.failed++
 	}
 }
+
+// a librarian collates dictionaries.
+type librarian struct {
+	rules map[string]string
+}
+
+// newLibrarian returns a new librarian populated with words and affix rules
+// obtained from the hunspell .dic file paths provided, checking that the
+// affix file aff also exists.
+func newLibrarian(aff, dic string) (librarian, error) {
+	_, err := os.Stat(aff)
+	if err != nil {
+		return librarian{}, err
+	}
+	l := librarian{make(map[string]string)}
+	err = l.addDictionary(dic)
+	if err != nil {
+		return librarian{}, err
+	}
+	return l, nil
+}
+
+// addDictionary adds word rules from the hunspell dictionary at the given
+// path.
+func (l librarian) addDictionary(path string) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	sc := bufio.NewScanner(f)
+	for i := 0; sc.Scan(); i++ {
+		if i == 0 {
+			// Skip word count line.
+			continue
+		}
+		err := l.addWord(sc.Text())
+		if err != nil {
+			return fmt.Errorf("%w at %s:%d", err, path, i+1)
+		}
+	}
+	return sc.Err()
+}
+
+// addWord adds the provided word to the librarian's dictionary merging any
+// affix rules into those already existing for the word.
+func (l librarian) addWord(w string) error {
+	r := strings.Split(w, "/")
```

Comment:
neat use case for strings.Cut :)

Reply:
I've been looking for that my whole life (well at least I wanted it last week).

Reply:
Though here I actually do want split so I can validate the word rule syntax.